cmd/dep/status.go

```diff
@@ -865,23 +1024,31 @@ func runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceMana
 			}
 		}
 
-		// A map of ProjectRoot and *BasicStatus. This is used in maintain the
-		// order of BasicStatus in output by collecting all the BasicStatus and
-		// then using them in order.
-		bsMap := make(map[string]*BasicStatus)
-		for bs := range bsCh {
-			bsMap[bs.ProjectRoot] = bs
-		}
+		if cmd.detail {
```

Comment:
Hi @zjs,

I would include `return false, errCount, err` at the end of this `if cmd.detail {` block and then get rid of the `else` block below and return again at the end. This removes a level of indentation without making things any more difficult to follow.

Reply:
Not a huge deal though. Just my personal preference. Neither way is more right than the other.

Reply:
I can see the appeal of that, but I think I'd like to leave this as-is for now; I think the right improvement here is to refactor this to remove some of the duplicated concepts (the 11 lines each of the `if`/`else` blocks seem way too similar to me, but I haven't figured out a good way to tidy that up).